### ISSUE TITLE
Swagger import - Security definition  and Schemes ignored

### ISSUE THREAD

**Issue by RitwikChatterjee** [a85/POSTMan-Chrome-Extension#1470]
Hi,
When I am importing Swagger definition into POSTMAN, the securitydefinition is ignored.
Also, I am not able to find all the schemes that I am defining.
FYI, here is what I have used:

schemes:
- http
- https
  securityDefinitions:
  basicAuth:
    type: basic
    description: HTTP Basic Authentication.

**Comment by abhijitkane** [comment]
@RitwikChatterjee The first part (scheme - https) has been fixed, and we'll push it in the next release.

Support for Basic Authentication is not present yet, we'll add it in a future release (can't offer an ETA on it though)

**Comment by theManikJindal** [comment]
@abhijitkane I face the same issue when importing a swagger with `securityDefinition` type `oauth2`. There is no Auth imported at all in any of my APIs even though the `security` object is present on each of them. 

Should I log a separate issue?

**Comment by TheCodeDestroyer** [comment]
Is it possible to have anything in schemes I would like to import that as a variable.

Example:

```
schemes:
 - "{{protocol}}"
```

I know this is not a valid swagger file but it's easier for me to change it here than after the import for 100+ endpoints

**Comment by theManikJindal** [comment]
@TheCodeDestroyer I think that would be a separate feature request. Maybe you would want to add a new issue for that.

**Comment by coolbung** [comment]
Do you support the security definitions yet ? I have the following securityDefinition in my swagger file
```yaml
securityDefinitions:
  Bearer:
    type: apiKey
    name: Authorization
    in: "header"
```

but postman seems to be ignoring it when importing the swagger file.

**Comment by liemle3893** [comment]
Any work on this issue?
I'm still having problem with Swagger Security.
```yaml
swagger: "2.0"
.
.
securityDefinitions:
  APIKeyQuery:
    type: apiKey
    in: query
    name: auth_key
  APIKeyHeader:
    type: apiKey
    in: header
    name: Authorization
```
Thanks!

**Comment by meantux** [comment]
Multiple API Keys' not really supported
In my example below you'll find a security defined as Swagger documented. Please note the second item does not have a dash, this is to indicate that they are both needed for the query. When I import it into Postman only the first security key is visible.

```
security:
  - PNClientId: []
    PNToken: []

components:
  securitySchemes:
    PNClientId:
      type: apiKey
      in: header
      name: x-client-id
    PNToken:
      type: apiKey
      in: header
      name: x-authentication-token

```

**Comment by IanKemp** [comment]
Importing the following schema as a collection results in all API operations being set to "Inherit auth from parent", but the collection's authorization is set to "No auth" (it should use API Key with the specified header name):

```json
{
  "components": {
    "securitySchemes": {
      "api": {
        "type": "apiKey",
        "description": "Authorize with your API key.",
        "name": "X-Api-Key",
        "in": "header"
      }
    }
  },
  "security": [
    {
      "api": [ ]
    }
  ]
}
```

Importing the following schema as a collection results in all API operations being set to "No auth", with a header of "api" added to every applicable operation, and the collection's authorization set to "No auth". Again, the security scheme should be applied to the whole collection, and since the operation's security matches the scheme, it should **not** declare an identical header but should inherit that scheme from the collection:

```json
{
  "paths": {
    "/api/v1/foo": {
      "get": {
        "security": [
          {
            "api": [ ]
          }
        ]
      }
    }
  },
  "components": {
    "securitySchemes": {
      "api": {
        "type": "apiKey",
        "description": "Authorize with your API key.",
        "name": "X-Api-Key",
        "in": "header"
      }
    }
  }
}
```

When will this incredibly basic functionality, AKA implementing the OpenAPI 2.0 spec that was published **over half a decade ago**, be implemented?

**Comment by apham-sy** [comment]
I am still waiting on this feature as the auto generated docs require manual intervention in order to be seamlessly used by our users. It's producing a identical header `x-api-key`. I would at least think that's a simple fix to not generate an identical header if it's already been defined in the securitySchema.

**Comment by VShingala** [comment]
@coolbung @IanKemp @apham-sy  We have fixed the mentioned issue in App v7.33. Try it out and feel free to report any problem with the fix.

Now conversion will not add individual header in the collection and use Authorization of request instead. You will see Authorization set to defined key at the global or request level depending upon where Security Requirement Object is defined. Furthermore, as scopes are defined for Authorization, having the same security requirement object at the Global level and at the Request level will not add Duplicate headers/queries.

Also as Authorization is set to API Key for request/collection, Documentation will reflect the same under Authorization.

@meantux We don't support multiple security schema at same time, You can follow https://github.com/postmanlabs/postman-app-support/issues/7427 for future updates.

**Comment by j-bjorne** [comment]
I am on Postman v8.0.6 and when I import a collection with:
```
,  
  "auth": {
    "type": "apikey",
    "apikey": [
      {
        "key": "value",
        "value": "{{api-key}}",
        "type": "string"
      },
      {
        "key": "key",
        "value": "api-key",
        "type": "string"
      }
    ]
  },
```

I still need to manually go into the collection and change the key and value as they are empty upon importing. You said this was fixed in release v7.33, is it somehow broken again or is it supposed to act this way?

**Comment by VShingala** [comment]
@j-bjorne I am not sure what are you asking, so what you have defined is the Auth part of collection and what issue mention is generated collection from OAS specification.

Still, considering what you mentioned is the authorization object of collection, I was able to import it correctly. [Here](https://github.com/VShingala/api-schema/blob/master/apikey_auth_pm_collection.json#L370) is sample collection I imported. and It did have the following Authorization as part of collection. (It's at the collection level and not at request level)

![image](https://user-images.githubusercontent.com/16756296/109645355-bce1ae00-7b7c-11eb-984f-7f460e5e06f9.png)

**Comment by habibur4340** [comment]
Hey @RitwikChatterjee  ! It seems like the security definitions and schemes in your Swagger import aren't being picked up correctly. This can happen for a few reasons:

**Version Compatibility:** Double-check if your Swagger file's version aligns with the tool's supported version.

**Syntax & Structure Check:** Minor errors in syntax or structure can lead to such issues. Validating your file against the Swagger spec might help.

**Tool-Specific Requirements:** Some tools have unique requirements or limitations for Swagger imports. It's worth looking into the specific documentation of the tool you're using.

If these steps don't resolve the issue, I recommend trying out [Apidog](https://apidog.com/). It's great for handling various API specifications and could provide a smoother import process for Swagger files, especially with security details. Hope this helps!"
